### PR TITLE
implementing some basics REAL

### PR DESCRIPTION
**Pull request by Sillyvan**

### CODE REVIEW

File: internal/handlers/starboard.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+			rng := rand.New(rand.NewSource(time.Now().UnixNano()))
+			index := rng.Intn(len(starboardMessages))
+
+			randomMessage := starboardMessages[index].MessageContent
+			speechBubbleComponent := templates.SpeechBubble(randomMessage)
+			err := speechBubbleComponent.Render(r.Context(), w)
+			if err != nil {
+				http.Error(w, err.Error(), http.StatusInternalServerError)
+				return
```
Comment (JakeSteam) [inline internal/handlers/starboard.go:40]: nit: Redundant

File: cmd/backend/main.go [hunk 2]
```diff
@@ -40,15 +48,27 @@ func main() {
...
 	}
 
 	if err := server.ListenAndServe(); !errors.Is(err, http.ErrServerClosed) {
 		slog.Error(err.Error())
 	}
 }
+
+func initDatabase() *gocqlx.Session {
+	cluster := gocql.NewCluster("127.0.0.1")
+	cluster.Keyspace = "discord"
```
Comment (minz1) [inline cmd/backend/main.go:66]: We should configure these with environment variables. I have a pattern for this in one of my other projects. Here's a code snippet.

```go
package pgconfig

import (
	"fmt"
	"log/slog"
	"os"
	"sort"
)

type PgConfig struct {
	user, database, host, password, port string
	sslMode string // optional
}

func PgConfigFromEnv() (PgConfig, error) {
	var missing []string

	get := func(key string) string {
		val := os.Getenv(key)
		if val == "" {
			missing = append(missing, key)
		}
		return val
	}

	cfg := PgConfig{
		user: get("PG_USER"),
		database: get("PG_DATABASE"),
		host: get("PG_HOST"),
		password: get("PG_PASSWORD"),
		port: get("PG_PORT"),
		sslMode: os.Getenv("PG_SSLMODE"), // optional
	}

	switch cfg.sslMode {
	case "", "disable", "allow", "require", "verify-ca", "verify-full":
	default:
		return cfg, fmt.Errorf(`invalid sslmode %s, expected one of: "", "disable", "allow", "require", "verify-ca", or "verify-full"`, cfg.sslMode)
	}

	if len(missing) > 0 {
		sort.Strings(missing)
		return cfg, fmt.Errorf("missing required environment variables: %v", missing)
	}

	return cfg, nil
}

func (pg PgConfig) String() string {
	s := fmt.Sprintf("postgres://%s:%s@%s:%s/%s", pg.user, pg.password, pg.host, pg.port, pg.database)

	slog.Info(s)

	if pg.sslMode != "" {
		s += "?sslmode=" + pg.sslMode
	}

	return s
}
```

Should be easy to adapt for Scylla.

**Review by minz1 — CHANGES_REQUESTED** [review]

File: cmd/backend/main.go [hunk 3]
```diff
@@ -21,22 +21,26 @@ func main() {
...
-	indexTemplate := templates.Index("2so cool!!")
-	indexEndpoint := handlers.CreateIndexEndpoint(indexTemplate)
-
 	db := initDatabase()
 	defer db.Close()
 
+	indexTemplate := templates.Index("2so cool!!")
+	indexEndpoint := handlers.CreateIndexEndpoint(indexTemplate)
+	UiEndpoint := handlers.CreateUiEndpoints(*db)
+	APIEnpoints := handlers.CreateAPIEndpoints(*db)
```
Comment (minz1) [inline cmd/backend/main.go:30]: these should be lowercase

File: internal/handlers/api.go [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+					http.Error(w, fmt.Sprintf("400 Bad Request: %v", err), http.StatusBadRequest)
+					log.Println(err)
+					return
+				}
+
+				query := db.Query(models.DiscordStarboard.Insert()).BindStruct(starboardMessage)
+
+				if err := query.ExecRelease(); err != nil {
+					http.Error(w, "500 Internal Server Error", http.StatusInternalServerError)
+					log.Println(err)
```
Comment (minz1) [inline internal/handlers/api.go:32]: use our slog here

**Pull request merged**